script.artistslideshow/resources/common/fileops.py

```diff
@@ -0,0 +1,208 @@
+            try:
+                _delete( os.path.join( src, '.DS_Store' ) )
+            except IOError:
+                log_lines.append( 'unable to delete .DS_Store file' )
+            except Exception as e:
+                log_lines.append( 'unknown error while attempting to delete .DS_Store file' )
+                log_lines.append( e )
+            _action = _rmdir
+        else:
+            _action = _delete
+        try:
+            log_lines.append( 'deleting %s %s' % (type, src) )
+            if isXBMC:
+                if not _action( src ):
+                    raise IOError( 'unable to delete item' )
+            else:
+                _action( src )                
+        except IOError:
+            log_lines.append( 'unable to delete %s' % src )
+            return False, log_lines
+        except Exception as e:
+            log_lines.append( 'unknown error while attempting to delete %s' % src )
+            log_lines.append( e )
+            return False, log_lines
+        return True, log_lines
+    else:
+        log_lines.append( '%s does not exist' % src )
+        return False, log_lines
+
+
+def moveFile( src, dst ):
+    log_lines = []
+    cp_loglines = []
+    dl_loglines = []
+    success = False
+    if _exists( src ):
+        cp_success, cp_loglines = copyFile( src, dst )
+        if cp_success:
+            dl_success, dl_loglines = deleteFile( src )
+            if dl_success:
+                success = True
+    else:
+        log_lines.append( '%s does not exist' % src)
+        success = False
+    return success, log_lines + cp_loglines + dl_loglines
+
+
+def atoi( text ):
+    return int(text) if text.isdigit() else text
+
+
+def naturalKeys( text ):
+    '''
+    alist.sort( key=naturalKeys ) sorts in human order
+    http://nedbatchelder.com/blog/200712/human_sorting.html
+    '''
+    return [ atoi( c ) for c in re.split( r'(\d+)', text ) ]
+
+
+def popenWithTimeout( command, timeout ):
+    log_lines = []
+    log_lines.append( 'running command ' + command)
+    if hasSubprocess:
+        try:
+            p = subprocess.Popen( command, stdout=subprocess.PIPE, stderr=subprocess.PIPE )
+        except OSError:
+            log_lines.append( 'error finding external script, terminating' )
+            return False, log_lines
+        except Exception as e:
+            log_lines.append( 'unknown error while attempting to run %s' % command )
+            log_lines.append( e )
+            return False, log_lines
+        for t in _range( timeout * 4 ):
+            time.sleep( 0.25 )
+            if p.poll() is not None:
+                return p.communicate(), log_lines
+        p.kill()
+        log_lines.append( 'script took too long to run, terminating' )
+        return False, log_lines
+    else:
+        log_lines.append( 'running command with os.system' )
+        os.system( command )
```

Comment:
I'm not the biggest fan of this. Also the same for the mv and delete generics. Makes it harder to review since we need to take into account all the usages of the methods

Reply:
I understand the review concern, but all the common libraries hold things that I use in many different projects (both Kodi addons and generic python scripts), and they allow me to include error checking that would otherwise have to be copied and pasted throughout the code (which would make it harder for me to manage).  I've been using these libraries for a number of years with Kodi submissions without issue, but if I have to get rid of them I will.  It'll just be awhile before I can resubmit since I'll have to rewrite a pretty substantial portion of AS.